src/teacher/views.py

```diff
@@ -1,113 +1,127 @@
 from django.db.models import Q
-from django.http import HttpResponse, Http404, HttpResponseBadRequest, HttpResponseRedirect, HttpResponseNotFound
-from django.shortcuts import render, redirect, get_object_or_404
+from django.http import HttpResponse, Http404
+from django.shortcuts import render, redirect
 
 # Create your views here.
-from django.urls import reverse
+from django.urls import reverse, reverse_lazy
+from django.views.generic import TemplateView, ListView, UpdateView, CreateView, DeleteView, DetailView
 
 from teacher.forms import TeacherAddForm, TeacherEditForm
+from teacher.mixins import TeacherValidFormMixin
 from .models import Teacher
 
 
-def index(request):
+class IndexTeacherDetailView(DetailView):
+    model = Teacher
+    template_name = 'teacher/index.html'
+    context_object_name = 'teacher'
 
-    id_teacher = request.GET.get('id')
-    if id_teacher is not None:
-        if not id_teacher.isdigit():
-            return HttpResponse("'id' must be numeric and greatest when zero")
+    def get_object(self, queryset=None):
 
-        id_teacher = int(id_teacher)
-        if id_teacher == 0:
-            return HttpResponse("'id' must be greatest when zero")
+        request = self.request
 
+        id_teacher = request.GET.get('id')
```

Comment:
It's better to use more high-level source of data as request params are already parsed by this moment

```
id_teacher = request.GET.get('id')
```
-->
```
id_teacher = self.kwargs.get(self.pk_url_kwarg, None)
```